specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json

```diff
@@ -0,0 +1,380 @@
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/HanaInstance"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "Resource": {
+            "description": "The resource model definition.",
+            "properties": {
+                "id": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "Resource ID"
+                },
+                "name": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "Resource name"
+                },
+                "type": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "Resource type"
+                },
+                "location": {
+                    "type": "string",
+                    "description": "Resource location"
+                },
+                "tags": {
+                    "type": "object",
+                    "additionalProperties": {
+                        "type": "string"
+                    },
+                    "description": "Resource tags"
+                }
+            },
+            "x-ms-azure-resource": true
+        },
+        "HanaInstancesListResult": {
+            "properties": {
+                "value": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/HanaInstance"
+                    },
+                    "description": "The list of SAP HANA on Azure instances."
+                },
+                "nextLink": {
+                    "type": "string",
+                    "description": "The URL to get the next set of HANA instances."
+                }
+            },
+            "description": "The response from the List HANA Instances operation."
+        },
+        "HanaInstance": {
+            "properties": {
+                "properties": {
+                    "x-ms-client-flatten": true,
+                    "$ref": "#/definitions/HanaInstanceProperties",
+                    "description": "HANA instance properties"
+                }
+            },
+            "allOf": [
+                {
+                    "$ref": "#/definitions/Resource"
+                }
+            ],
+            "description": "HANA instance info on Azure (ARM properties and HANA properties)"
+        },
+        "HanaInstanceProperties": {
+            "properties": {
```

Comment:
can any property in here be marked as readOnly?
Also applies to `HardwareProfile`'s `hardwareType`, some `OSProfile` properties, `NetworkProfile` and `IpAddress`.

Reply:
Since this is currently going to be a readOnly spec, I've made all the fields readOnly.